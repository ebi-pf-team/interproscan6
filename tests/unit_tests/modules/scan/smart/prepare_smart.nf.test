nextflow_process {

    name "Test Process PREPARE_SMART"
    script "modules/smart/main.nf"
    process "PREPARE_SMART"

    stage {
        symlink "tests/data/channels/smart.hmmer3.hmmsearch.out"
        symlink "tests/data/channels/smart.hmmer3.faa"
        symlink "tests/data/databases/smart/9.0"
    }

    test("Should run PREPARE_SMART without failures") {

        when {
            process {
                """
                input[0] = [1, file("meta/tests/data/channels/smart.hmmer3.hmmsearch.out"), file("meta/tests/data/channels/smart.hmmer3.faa")]
                input[1] = "meta/tests/data/databases/smart/9.0/"
                input[2] = "hmmer"
                input[3] = 1000
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
