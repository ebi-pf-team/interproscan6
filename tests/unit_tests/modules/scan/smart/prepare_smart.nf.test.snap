{
    "Should run PREPARE_SMART without failures": {
        "content": [
            {
                "0": [
                    [
                        1,
                        "chunk_0.fasta:md5,5546ca8e3faf803b0f4b7c7ebc3d8468",
                        [
                            "SM00382",
                            "SM00826",
                            "SM00534",
                            "SM00533"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-02T11:53:03.568958"
    }
}